### ISSUE TITLE
Feature Request: Possibility to generate new key

### ISSUE THREAD

**Issue by sangfroid70** [SilenceIM/Silence#585]
### Feature Request
After problems with my phone and differents versions of Silence, I imported an encrypted backup and I lost my key. I thought I could generate new one but I realized that it was not possible.

Why don't implement such a function ?

Thank's for the work !!

### How to reproduce
- Not applicable

### Device info
- Device: Anyone

**Comment by ArchangeGabriel** [comment]
What makes you think you cannot generate a new one?

**Comment by sangfroid70** [comment]
Your question is interesting ;) I did not find any way to renew my key without delete application's datas. If there's a way, I would be happy to know it

**Comment by ArchangeGabriel** [comment]
Oh, you’re talking of the main key. I thought you were talking of session keys (which can be changed by just stopping/starting an encrypted session), but that was clearly a misunderstanding from my part.

I’m not sure how you can lose it but not everything else, so that you’re still not able to reset app data. Once key is lost, the only thing that I see remaining are the messages, but those can be exported in plain text (excepted for MMS, see #140 and #184) before resetting the app.

I’m not sure too why would someone want to change their main key (and thus whether this feature should be provided).

**Comment by pR0Ps** [comment]
Do you mean the passphrase that your messages are encrypted with?

If you mean the main identity key, I'm not sure how you managed to get in a state where it was deleted. The reason there is no way to regenerate a main key is that there's no way you're supposed to be able to delete it.

**Comment by sangfroid70** [comment]
Yes, I was speaking about the main identity key. The reason why I lost it is probably a bug when I upgraded in 0.16.3. An option to regenerate it would have been so usefull. I had to export in plaintext my messages, clear the application's data and import them. But, I lost my MMS. It's not dramatic but I've been surprised not to find the option.

Thank's for your answers, et excuse my poor english ;)
